From 044b55556675b0f8737c0ab1fb2ec139dcdc69af Mon Sep 17 00:00:00 2001
From: LI Qingwu <Qing-wu.Li@leica-geosystems.com.cn>
Date: Thu, 10 Oct 2024 12:56:58 +0800
Subject: [PATCH 4/4] WPE Platform: fix wpe_view_resize was not declared issue

fix Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp:87:5:
error: 'wpe_view_resize' was not declared in this scope;
did you mean 'wpe_view_resized'?

Upstream-Status: Submitted [https://github.com/WebKit/WebKit/pull/34441]

Signed-off-by: LI Qingwu <Qing-wu.Li@leica-geosystems.com.cn>
---
 Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp b/Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp
index 5ff20ab8c2b8..d0d5509c1582 100644
--- a/Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp
+++ b/Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp
@@ -84,7 +84,9 @@ void WPEQtView::geometryChange(const QRectF& newGeometry, const QRectF&)
         return;
 
     auto* wpeView = webkit_web_view_get_wpe_view(d->m_webView.get());
-    wpe_view_resize(wpeView, d->m_size.width(), d->m_size.height());
+    if (auto* wpeToplevel = wpe_view_get_toplevel(wpeView))
+        wpe_toplevel_resize(wpeToplevel, d->m_size.width(), d->m_size.height());
+
 }
 
 void WPEQtView::configureWindow()
@@ -119,7 +121,9 @@ void WPEQtView::createWebView()
         "settings", settings.get(), nullptr)));
 
     auto* wpeView = webkit_web_view_get_wpe_view(d->m_webView.get());
-    wpe_view_resize(wpeView, d->m_size.width(), d->m_size.height());
+    if (auto* wpeToplevel = wpe_view_get_toplevel(wpeView))
+        wpe_toplevel_resize(wpeToplevel, d->m_size.width(), d->m_size.height());
+
     wpe_view_map(wpeView); // FIXME: unmap when appropriate and implement can_be_mapped if needed.
 
     if (!wpe_view_qtquick_initialize_rendering(WPE_VIEW_QTQUICK(wpeView), this, &error.outPtr())) {
-- 
2.34.1

